Results rewrite part2 (#61)
greenelab/text_mined_hetnet_manuscript: content/04.results.md

The y-axis shows the area under the receiver operating curve (AUROC).
Each point on the plot shows the average of 50 sample runs, while the error bars show the 95% confidence intervals of all runs.
The baseline and “All” data points consist of sampling from the entire fixed set of label functions.
](https://raw.githubusercontent.com/danich1/snorkeling-full-text/cd38c26db62f7eb7bc83fd9c424d0c8912512d06/figure_generation/output/figure_four.png){#fig:auroc_grabbag_gen_model_test_set}

### Discriminative Model Performance

The discriminative model is intended to augment performance over the generative model by incorporating textual features together with estimated training labels.
We found that the discriminative model generally outperformed the generative model with respect to AUROC as more edge-specific label functions were incorporated (Figure {@fig:auroc_discriminative_model_performance}).
Regarding AUPR, this model outperformed the generative model for the Disease-associates-Gene (DaG) edge type. 
At the same time, it had close to par performance for the rest of the edge types (Supplemental Figure {@fig:aupr_discriminative_model_performance}).

> The discriminative model's performance was often poorest when very few edge-specific label functions were incorporated into the baseline model (seen in DaG, Compound-binds-Gene (CbG), and Gene-interacts-Gene (GiG)). 

Reviewer: ```suggestion
The discriminative model's performance was often poorest when very few edge-specific label functions were incorporated into the baseline model (seen in DaG, CbG, and Gene-interacts-Gene (GiG)). 
```
You already defined this above. Just before we submit, you should check to see that these are defined one and only one time and then used from there.

The y-axis shows the area under the receiver operating curve (AUROC).
Each point on the plot shows the average of 50 sample runs, while the error bars show the 95% confidence intervals of all runs.
The baseline and “All” data points consist of sampling from the entire fixed set of label functions.
](https://raw.githubusercontent.com/danich1/snorkeling-full-text/cd38c26db62f7eb7bc83fd9c424d0c8912512d06/figure_generation/output/figure_four.png){#fig:auroc_grabbag_gen_model_test_set}

### Discriminative Model Performance

> The discriminative model augments performance over the generative model by incorporating textual features together with estimated training labels.

Reviewer: ```suggestion
The discriminative model is intended to augment performance over the generative model by incorporating textual features together with estimated training labels.
```

> We found that the discriminative model generally outperforms the generative model with respect to AUROC as more edge-specific label functions are incorporated (Figure {@fig:auroc_discriminative_model_performance}).

Reviewer: When reporting actual results, it should be past tense. Probably good to revise this paragraph to fit.

The recall for the Hetionet v1 knowledge graph is shown as a percentage in parentheses.
For example, for the Compound treats Disease (CtD) edge, our method recalls 32% of existing edges and can add 11,146 new ones.
](https://raw.githubusercontent.com/danich1/snorkeling-full-text/cd38c26db62f7eb7bc83fd9c424d0c8912512d06/figure_generation/output/figure_eight.png){#fig:hetionet_reconstruction}

One of the goals of our work is to measure the extent to which learning multiple edge types could construct a biomedical knowledge graph.
Using Hetionet v1 as an evaluation set, we measured this framework's recall and quantified the number of edges that may be incorporated with high confidence.
Overall, we were able to recall about thirty percent of the preexisting edges for all edge types (Figure {@fig:hetionet_reconstruction}) and report our top ten scoring sentences for each edge type in Supplemental Table {@tbl:edge_prediction_tbl}.
Our best recall is with the Disease associates Gene (DaG) and Compound binds Gene (CbG) edge types, where we retain 33% of preexisting edges.

> Plus, we can add over 100,000 new edges into both categories.

Reviewer: ```suggestion
Our analysis was able to supplement the existing network with 100,000 additional edges.
```
Is this at some probability threshold?
